experimental/gpu_monitoring/gpu_data_collector.py

```diff
@@ -303,12 +313,72 @@ def get_nfs_rates(nfs_counters, time_interval_seconds, hostname, physicalhostnam
             nfs_counters[mount_pt]["client_read_iop"] = current_nfs_counters[mount_pt]["client_read_iop"]
             nfs_counters[mount_pt]["client_write_iop"] = current_nfs_counters[mount_pt]["client_write_iop"]
             nfs_counters[mount_pt]["client_write_bytes"] = current_nfs_counters[mount_pt]["client_write_bytes"]
-    nfs_rates['hostname'] = hostname
-    nfs_rates['physicalhostname'] = physicalhostname_val
 
     return nfs_rates_l
 
 
+def read_file(file_path):
+    f = open(file_path, "r")
+    file_lines_l = f.readlines()
+
+    return file_lines_l
+
+
+def find_value_in_file(find_string, index, file_lines_l):
+    value = 0
+    for line in file_lines_l:
+        if line.find(find_string) >= 0:
+            value = line.split()[index]
+            break
+    return int(value)
+
+
+def get_cpu_mem_data(hostname, physicalhostname_val, have_jobid, slurm_jobid):
+    cpu_mem_l = []
+    cpu_mem_d = {}
+    meminfo_l = read_file("/proc/meminfo")
+    for cpu_mem_counter_name in CPU_MEM_COUNTERS:
+        value = find_value_in_file(cpu_mem_counter_name, 1, meminfo_l)
+        cpu_mem_d[cpu_mem_counter_name + "_KB"] = value
+    if have_jobid:
+        cpu_mem_d['slurm_jobid'] = slurm_jobid
+    cpu_mem_d['hostname'] = hostname
+    cpu_mem_d['physicalhostname'] = physicalhostname_val
+    cpu_mem_l.append(cpu_mem_d)
+
+    return cpu_mem_l
+
+
+def get_cpu_loadavg_data():
+    cpu_loadavg_l = []
+    cpu_loadavg_l = read_file("/proc/loadavg")
+    value = cpu_loadavg_l[0].split()[0]
+
+    return value
+
+
+def get_cpu_data(hostname, physicalhostname_val, have_jobid, slurm_jobid):
```

Comment:
Thoughts about converting frequency to seconds?
The number of clock ticks per second can be obtained via `getconf CLK_TCK`.

Reply:
I prefer to dump the raw data to log analytics (for speed) and do the conversion (whatever unit is of interest) in the log analytics kusto query. I think this conversion is fairly straightforward with USER_HZ = 1/100 sec.